### PR TITLE
 fix: avoid using prefixIdenntifier in esm

### PR DESCRIPTION
**Pull request by elevatebart**
When using VTU within vite, the version used is esm-bundler. 
This version imports the esm-bundler of the compiler-core.

This version does not support prefixIdentifier.

I tried to make a conditional build of VTU.

### CODE REVIEW

File: src/utils/compileSlots.ts [hunk 1]
```diff
@@ -15,10 +15,13 @@ export function processSlot(source = '', Vue = vue) {
     `<SlotWrapper v-bind="$attrs">${template}</SlotWrapper>`,
     {
       mode: 'function',
-      prefixIdentifiers: true
+      prefixIdentifiers: __BROWSER__
```
Comment (lmiller1990) [inline src/utils/compileSlots.ts:18]: I think you may need to update tsconfig (or somewhere else?) to make this compile.

**Review by lmiller1990 — COMMENTED** [review]
Cool, once this is approved and all passing, we can merge.

**Comment by lmiller1990** [comment]
Thanks @elevatebart, is there some way I can test this out to make sure it's going to work how we want it to? Do you have some way of testing (assuming you did with Vite etc)?

**Comment by elevatebart** [comment]
We could create a playground-like environment to run VTU in a vite/ESM environment.
I think I would like to call the environment variable with another name than `__BROWSER__`. Maybe __ESM__ ? I called it this way because it is called this way in vue-next but I find it confusing. 
We use `__BROWSER__` in node and `!__BROWSER__` in vite with in-browser testing.
 
The good news is there is no emergency right now as I found an "unclean" workaround.

**Comment by lmiller1990** [comment]
I will release this within 24h.

**Comment by elevatebart** [comment]
thank you !!!

**Pull request merged**